### PR TITLE
create Socials Feed component

### PR DESCRIPTION
**Pull request by Evkos**

### CODE REVIEW

File: framework/Supports/SocialsFeed.php [hunk 1]
```diff
@@ -0,0 +1,360 @@
...
+			$timestamp     = strtotime( $fb_post['created_time'] );
+			$this->posts[] = array(
+				'post_title'   => 'Facebook post #' . $fb_post['id'],
+				'post_content' => ( ! empty( $fb_post['message'] ) ) ? $fb_post['message'] : '&nbsp;',
+				'post_date'    => date( 'Y-m-d H:i:s', $timestamp ),
+				'post_name'    => 'facebook_post_' . $fb_post['id'],
+				'post_type'    => 'socials_feed',
+				'tax_input'    => array( 'socials_category' => array( 'facebook' ) ),
+				'meta_fields'  => array(
+					'postmeta_image' => ( ! empty( $fb_post['full_picture'] ) ) ? $fb_post['full_picture'] : '',
```
Comment (ksamojlenko) [inline framework/Supports/SocialsFeed.php]: why you don't save link to original post?

File: framework/Supports/SocialsFeed.php [hunk 2]
```diff
@@ -0,0 +1,360 @@
...
+			$tab->createOption( array(
+				'name' => 'Instagram API Secret',
+				'id'   => 'instagram_api_secret',
+				'type' => 'text',
+			) );
+
+			$tab->createOption( array(
+				'name' => 'Instagram Access Token',
+				'id'   => 'instagram_access_token',
+				'type' => 'text',
```
Comment (ksamojlenko) [inline framework/Supports/SocialsFeed.php]: add hint, how to obtain access token

File: framework/Supports/SocialsFeed.php [hunk 3]
```diff
@@ -0,0 +1,360 @@
...
+			if ( ! isset( $media->data ) || ( isset( $media->code ) && in_array( $media->code, [
+						'400',
+						'401',
+						'403',
+						'404',
+						'500',
+					] ) )
+			) {
+				delete_post_meta( get_the_ID(), 'instagram_access_token' );
+				$sign_in = $instagram->getLoginUrl();
```
Comment (ksamojlenko) [inline framework/Supports/SocialsFeed.php]: from where you get get_the_ID()?

File: framework/Supports/SocialsFeed.php [hunk 4]
```diff
@@ -0,0 +1,360 @@
...
+			} else {
+				$insta_posts = $media->data;
+			}
+		}
+
+		foreach ( $insta_posts as $insta_post ) {
```
Comment (ksamojlenko) [inline framework/Supports/SocialsFeed.php]: insta_posts - incorrect naming of variable, and set variable before condition

File: framework/Supports/SocialsFeed.php [hunk 5]
```diff
@@ -0,0 +1,360 @@
...
+			$timestamp     = $insta_post->created_time;
+			$this->posts[] = array(
+				'post_title'   => 'Instagram post #' . $insta_post->id,
+				'post_content' => ( ! empty( $insta_post->caption->text ) ) ? $insta_post->caption->text : '&nbsp;',
+				'post_date'    => date( 'Y-m-d H:i:s', $timestamp ),
+				'post_name'    => 'instagram_post_' . $insta_post->id,
+				'post_type'    => 'socials_feed',
+				'tax_input'    => array( 'socials_category' => array( 'instagram' ) ),
+				'meta_fields'  => array(
+					'postmeta_image' => ( ! empty( $insta_post->images->standard_resolution->url ) ) ? $insta_post->images->standard_resolution->url : '',
```
Comment (ksamojlenko) [inline framework/Supports/SocialsFeed.php]: add origin post link to post meta fields

File: framework/Supports/SocialsFeed.php [hunk 6]
```diff
@@ -0,0 +1,360 @@
...
+			$this->posts[] = array(
+				'post_title'   => 'Twitter post #' . $twitter_post->id,
+				'post_content' => ( ! empty( $twitter_post->text ) ) ? $twitter_post->text : '&nbsp;',
+				'post_date'    => date( 'Y-m-d H:i:s', $timestamp ),
+				'post_name'    => 'twitter_post_' . $twitter_post->id,
+				'post_type'    => 'socials_feed',
+				'post_status'  => 'publish',
+				'tax_input'    => array( 'socials_category' => array( 'twitter' ) ),
+				'meta_fields'  => array(
+					'postmeta_image' => ( ! empty( $twitter_post->entities->media[0]->media_url ) ) ? $twitter_post->entities->media[0]->media_url : '',
```
Comment (ksamojlenko) [inline framework/Supports/SocialsFeed.php]: save origin post link

File: framework/Supports/SocialsFeed.php [hunk 7]
```diff
@@ -0,0 +1,360 @@
...
+
+		$this->get_fb_posts();
+
+		if ( class_exists( 'MetzWeb\Instagram\Instagram' ) ) {
+			$this->get_insta_posts();
+		}
+
+
+		$this->get_twitter_posts();
+
```
Comment (ksamojlenko) [inline framework/Supports/SocialsFeed.php]: why you checking class sexists only for Instagram?

File: framework/Supports/SocialsFeed.php (hunk 5 above)
Comment (ksamojlenko) [inline framework/Supports/SocialsFeed.php]: if it video you should save type of post for use instagram embed on frontend

File: framework/Supports/SocialsFeed.php [hunk 8]
```diff
@@ -0,0 +1,360 @@
...
+	/**
+	 * SocialsFeed constructor.
+	 */
+	public function __construct() {
+		add_action( 'init', array( $this, 'register_socials_posttype' ) );
+		add_action( 'init', array( $this, 'register_socials_taxonomy' ) );
+		add_action( 'init', array( $this, 'create_settings_panel' ) );
+
+		wp_schedule_event( time(), 'hourly', 'get_social_posts' );
```
Comment (ksamojlenko) [inline framework/Supports/SocialsFeed.php]: for wp cron scheduler you should use code like this

add_action('jtf_social_sheduler', array($this, 'get_social_posts'));
if (! wp_next_scheduled ( 'jtf_social_sheduler')) {
    wp_schedule_event(time(), 'hourly', 'jtf_social_sheduler');
}

File: framework/Supports/SocialsFeed.php [hunk 9]
```diff
@@ -0,0 +1,360 @@
...
+
+
+		add_action( 'init', array( $this, 'insert_posts' ) );
```
Comment (ksamojlenko) [inline framework/Supports/SocialsFeed.php]: you can merge all of init methods into one hook

File: framework/Supports/SocialsFeed.php (hunk 7 above)
Comment (ksamojlenko) [inline framework/Supports/SocialsFeed.php]: will be better move  'get xx posts' to the another method

File: framework/Supports/SocialsFeed.php [hunk 10]
```diff
@@ -0,0 +1,360 @@
...
+			'hierarchical' => false,
+		) );
+
+	}
+
+	/**
+	 * Create Titan Framework tab for socials API settings
+	 */
+	public function create_settings_panel() {
+
```
Comment (ksamojlenko) [inline framework/Supports/SocialsFeed.php]: in this method doesn't needed to check socialfeeds classes exists

File: framework/Supports/SocialsFeed.php [hunk 11]
```diff
@@ -0,0 +1,360 @@
...
+		add_action( 'init', array( $this, 'insert_posts' ) );
+
+	}
+
+	/**
+	 * Register Socials post type
+	 */
+	public function register_socials_posttype() {
+		register_post_type( 'socials_feed', array(
```
Comment (ksamojlenko) [inline framework/Supports/SocialsFeed.php]: create constant for post type name

File: framework/Supports/SocialsFeed.php [hunk 12]
```diff
@@ -0,0 +1,360 @@
...
+			'labels'       => array(
+				'name' => 'Socials Feed',
```
Comment (ksamojlenko) [inline framework/Supports/SocialsFeed.php]: add locale function for statick texts

File: framework/Supports/SocialsFeed.php [hunk 13]
```diff
@@ -0,0 +1,360 @@
...
+			'hierarchical' => true,
+			'taxonomies'   => array(),
+		) );
+	}
+
+	/**
+	 * Register Socials taxonomy
+	 */
+	public function register_socials_taxonomy() {
+		register_taxonomy( 'socials_category', array( 'socials_feed' ), array(
```
Comment (ksamojlenko) [inline framework/Supports/SocialsFeed.php]: create constant for taxonomy slug

File: framework/Supports/SocialsFeed.php (hunk 10 above)
Comment (ksamojlenko) [inline framework/Supports/SocialsFeed.php]: add options for disable obtain social posts gain by scheduler

File: framework/Supports/SocialsFeed.php [hunk 14]
```diff
@@ -0,0 +1,360 @@
...
+		return true;
+
+	}
+
+	protected function is_exists( $post_title ) {
+		return get_page_by_title( wp_strip_all_tags( $post_title ), OBJECT, 'socials_feed' );
+	}
+
+
+}
```
Comment (ksamojlenko) [inline framework/Supports/SocialsFeed.php]: add more comments for object methods

**Comment by aprokopenko** [comment]
deprecated

**Pull request closed without merging**